posts/2024-04-20-Performance Comparison: Data.table in R and Pandas in Python-Doris Amoakohene/index.qmd

```diff
@@ -0,0 +1,392 @@
+---
+title: 'Performance Comparison: Data.table in R and Pandas in Python'
+author: "Doris Amoakohene"
+date: "2024-03-03"
+output: html_document
+---
+
+# Introduction:
+
+R and Python are two programming languages that have gained immense popularity among data scientists, statisticians, and researchers. 
+In this blog, we will explore two widely used libraries, data.table in R and pandas in Python, which excel in data manipulation and provide versatile functionalities for working with data, focusing on their capabilities for reading, writing, and reshaping data.
+We will also explain how to graphically demonstrate the time taken by each operation. We will use atime package to compare and visualize the asymptotic performance (time and memory usage) of the different functions mentioned above.
+By comparing the asymptotic performance of these packages in these programing languages, we aim to provide insights into their usage and help data scientists make informed choices when it comes to data manipulation and analysis.
+
+
+In the atime::atime() we define the following arguments, 
+
+The first argument N is a sequence of data sizes.This explores computational efficiency, which will compute time/memory usage for different data sizes(N).
+
+The second argument setup is an expression that will be evaluated for each value in N, to create data of a given size.
+
+seconds.limit may optionally be specified. If an expression is slower than this limit for any data size, then no larger data sizes will be measured.
+
+Lastly, expr.list, which is the list of expressions for which time/memory usage will be measured.
+
+## Various plot
+
+atime::reference_best() : This plot shows the asymptotic reference lines, which defines the estimated asymptotic time and memory complexity. The output and figure will includes violet reference lines, in which the text labels can be interpreted in terms of big O notation (asymptotic time and memory usage).
+
+The predict() function is used to generate predictions based on the atime::reference_best() dataset. The resulting plot illustrates the data size, N, that can be processed within a specific time or memory limit.
+
+
+## Libraries
+
+```{r setup,warning=FALSE,message=FALSE}
+
+library(data.table)
+library(reshape2)
+library(atime)
+library(ggplot2)
+library(reticulate)
+use_python("C:/Users/amoak/AppData/Local/Programs/Python/Python312/python.exe")
```

Comment:
this will not be reproducible on another computer, so at least you should add a comment that says something like "If you want to reproduce, please change to the path of python on your computer"